### PR TITLE
[FFP-513] Implements LanguageDetectionService

### PR DESCRIPTION
**Pull request by antoniofpnmartins**
This pull request implements the language detection service.

- [x] Checked the work against _all_ of the ticket requirements
- [ ] CHANGELOG updated
- [x] Formatter has been run against all changed files
- [x] All changed files have been added correctly
- [ ] Appropriate unit tests and integration tests have been added
- [ ] All tests are passing
- [x] Ad hoc testing has been done
- [x] Affected docs have been updated
- [x] Tickets created for any TODOs
- [x] All licenses have been checked and confirmed for commercial use

### CODE REVIEW

File: api/src/recursiveai/findflow/assistant/api/services/language_detection.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            def _lambda() -> None:
+                model_path = os.path.join(
+                    self._config.local_directory, self._config.language_model_filename
+                )
+                self._model = load_model(model_path)
+
+            await self._helper.sync_to_async(_lambda)
+
+    async def detect_language(self, text: str) -> str:
+        await self.download_model()
```
Comment (miguelnmiranda) [inline api/src/recursiveai/findflow/assistant/api/services/language_detection.py:54]: Like with embedder this should be done on startup.

File: api/config/example/api_config.yaml [hunk 2]
```diff
@@ -43,4 +43,7 @@ assistants:
 embeddings:
   local_directory: "embedder_model"
 
+language_detection:
+  local_directory: "fasttext_models"
```
Comment (miguelnmiranda) [inline api/config/example/api_config.yaml:47]: How do we populate the local folders? also applies to embedder.

**Review by miguelnmiranda — APPROVED** [review]
LGTM otherwise

File: api/src/recursiveai/findflow/assistant/api/services/language_detection.py (hunk 1 above)
Comment (antoniofpnmartins) [reply api/src/recursiveai/findflow/assistant/api/services/language_detection.py:54]: It is, in the provider. However, if that's skipped for whatever reason, this ensures that the model is downloaded before being used. Note that it only runs once: after self._model is populated, download_model() won't do anything.

File: api/config/example/api_config.yaml (hunk 2 above)
Comment (antoniofpnmartins) [reply api/config/example/api_config.yaml:47]: Manually

Comment (miguelnmiranda) [reply api/config/example/api_config.yaml:47]: Please add the `how to` to readme, cloud commands or otherwise.

**Pull request merged**